### PR TITLE
Simulation extended example

### PR DESCRIPTION
**Pull request by danieljanes**

### CODE REVIEW

**Comment by danieljanes** [comment]
@jafermarq this PR contains your `simulation_ray` example. I've updated it to make use of the new `client_fn` API, would you mind double-checking if this is ready to be reviewed and merged?

**Comment by jafermarq** [comment]
@danieljanes , I just tested the latest commit in this PR and the example is working fine

**Comment by tanertopal** [comment]
> 
> 
> @danieljanes , I just tested the latest commit in this PR and the example is working fine. The `simulation_ray` might not be the best name I'd say. If there is an equivalent example for TF ( I remember you put together one some weeks ago), maybe we should give them the same name and just add the `_tensorflow` and `_pytorch` suffix?

I also think naming it `_pytorch` instead of `_ray` might be reasonable.

**Comment by danieljanes** [comment]
@jafermarq @tanertopal I've renamed the example to `simulation_pytorch` (and the prior `simulation_pytorch` to `simulation_pytorch_legacy`). With that, the example should be ready.

File: examples/simulation_pytorch/README.md [hunk 1]
```diff
@@ -1,69 +1,26 @@
-# Single-Machine Simulation of Federated Learning Systems
+# example_simulation_ray
 
-This example is part of a series of blog posts. It's recommended to read the [blog post](https://flower.dev/blog/2021-01-14-single-machine-simulation-of-federated-learning-systems) before reading further.
+This is the code splits CIFAR-10 dataset into `pool_size` partitions (user defined) and does a few rounds of CIFAR-10 training.
 
-## Quick Start
+## Requiremets
```
Comment (tanertopal) [inline examples/simulation_pytorch/README.md:5]:
```suggestion
## Requirements
```

File: examples/simulation_pytorch/README.md [hunk 2]
```diff
@@ -1,69 +1,26 @@
-# Single-Machine Simulation of Federated Learning Systems
+# example_simulation_ray
```
Comment (tanertopal) [inline examples/simulation_pytorch/README.md:1]: Shall we have a more descriptive title? E.g. `Single-Machine Simulation using simulation API` or similar?

File: examples/simulation_pytorch/README.md [hunk 3]
```diff
@@ -1,69 +1,26 @@
...
 
-This example is part of a series of blog posts. It's recommended to read the [blog post](https://flower.dev/blog/2021-01-14-single-machine-simulation-of-federated-learning-systems) before reading further.
+This is the code splits CIFAR-10 dataset into `pool_size` partitions (user defined) and does a few rounds of CIFAR-10 training.
```
Comment (tanertopal) [inline examples/simulation_pytorch/README.md:3]:
```suggestion
This code splits CIFAR-10 dataset into `pool_size` partitions (user defined) and does a few rounds of CIFAR-10 training.
```

**Comment by tanertopal** [comment]
Just added a few suggestions

**Review by tanertopal — APPROVED** [review]

**Pull request merged**